enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -45,43 +66,62 @@ Organizations desire to reuse their existing enterprise log solutions to store c
 * It is not a goal to provide secure storage for audit logs. If the deployer chooses to enable audit log forwarding, they need to make sure that the endpoint is compliant with governmental regulations and secure. The OpenShift logging Elasticsearch does not comply with those regulations.
+* The logging API will not expose fluentd configuration, it is intended to be independent of the underlying collector and output technologies.
 
 ## Proposal
 
-Log forwarding will provide a declarative way to specify the outputs for specific types of logs using a 'pipeline'.  A 'pipeline' defines simple routing for one source to one or more outputs. The source of logs are opinionated and well defined by cluster logging. The initial source types are as follows:
+Log forwarding will provide a declarative 'pipeline' that associates a set of container logs with a set of outputs. The pipeline selects source logs by:
+
+* Source type: Application, Infrastructure, Audit
+* K8s label selector maps and/or expressions.
+* Namespace (for global forwarders)
 
-* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
-* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
-* `logs.audit` - Logs generated by the nodes' auditd (/var/log/audit/audit.log), audit logs from the kubeapi-server and the openshift-apiserver. This will not be forwarded by default.
+The source types are:
 
-Future types may be:
-* `events.kubernetes` - Kubernetes events
+* `Application` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `Infrastrucure` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infrastructure" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+* `Audit` - Logs generated by the nodes' auditd (/var/log/audit/audit.log), audit logs from the kubeapi-server and the openshift-apiserver. This will not be forwarded by default.
+
+Future types may include:
+* `Events` - Kubernetes events
 
 There are no assumptions regarding whether or not an endpoint is deployed on or off cluster.  Endpoints off-cluster may require adminstrators to perform additional actions in order for logs to be forwarded (e.g. secret creation, opening port, enable global proxy configuration) 
 Following is the list of supported endpoint types for this proposal:
 
-* Elasticsearch (v6.x)
-* Fluent `forward`
+* Elasticsearch (v6.x) with/without TLS.
+* Fluent `forward` with/without TLS.
+* Syslog UDP, TCP, TLS.
+<!-- XXX(alanconway) What's the target set for GA? -->
```

Comment:
Do we need to be explicit here on how the certificate expiry and maybe rotation will affect the collector's pods? IMHO for external logging outputs we will not have control of expired server certificates. 
- Do we need to add metrics and alerts for the OKD admin to react upon on time?
- IIRC rotation of DaemonSet pods is an all or nothing op. Rotating one cert for one output will affect all things on transit. Correct?

Reply:
I think as with anything related to certs we don't control it is up to the cert owner to rotate their own certs.  Presumably when they expire you should start to see other things manifested as alerts because the endpdoint is down.  Rotation of DS pods are an all or nothing.  We will need to figure out a cadence of accepting LF changes so in order to not churn log collection

Reply:
we don't have to rotate them, but we need to react to the CAs/certs changing (e.g. if we have the admin provide them in a configmap, we need to watch that configmap for changes).

Reply:
Good point - we are using a SecretReference to refer to a secret object defined by the user, so we would need to watch those objects for changes. I'll mention that in the proposal.